addons/binding/org.openhab.binding.unifi/src/main/java/org/openhab/binding/unifi/internal/UniFiControllerConfig.java

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.unifi.internal;
+
+import java.math.BigDecimal;
+
+import org.apache.commons.lang.StringUtils;
+
+/**
+ *
+ * The {@link UniFiControllerConfig} encapsulates all the configuration options for an instance of the
+ * {@link UniFiController}.
+ *
+ * @author Matthew Bowman - Initial contribution
+ */
+public class UniFiControllerConfig {
+
+    private String host;
+    private BigDecimal port;
+    private String username;
+    private String password;
+    private BigDecimal refresh;
```

Comment:
When you replace `BigDecimal` with `int` you will not have to call `.intValue()` everywhere which will make the code more readable.

Reply:
@wborn are you sure, Afaik there only limited types supported by the configuration classes used via the ESH getConfiguration.as(...class)

Reply:
Yes the Configuration class is using [reflection logic](https://github.com/eclipse/smarthome/blob/6852de6b356f76e7d832e629034e7ea2b0d3c349/bundles/config/org.eclipse.smarthome.config.core/src/main/java/org/eclipse/smarthome/config/core/Configuration.java#L90) for this. I've used it myself this way e.g. in the [PlugwiseSenseConfig](https://github.com/openhab/openhab2-addons/blob/d3e80386d3b67a7c7c47b74fbeac1d0ca385a369/addons/binding/org.openhab.binding.plugwise/src/main/java/org/openhab/binding/plugwise/internal/config/PlugwiseSenseConfig.java#L31) class.

Reply:
Great, then there is some room for improvements in some bindings 😃